core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7679SingleMojoNoPomTest.java

```diff
@@ -0,0 +1,64 @@
+package org.apache.maven.it;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+
+import org.apache.maven.shared.verifier.Verifier;
+import org.apache.maven.shared.verifier.util.ResourceExtractor;
+import org.junit.jupiter.api.Test;
+
+/**
+ * This is a test set for <a href="https://issues.apache.org/jira/browse/MNG-7679">MNG-7679</a>.
+ * Executing single mojo without POM should not NPE.
+ *
+ */
+class MavenITmng7679SingleMojoNoPomTest
+    extends AbstractMavenIntegrationTestCase
+{
+
+    public MavenITmng7679SingleMojoNoPomTest()
+    {
+        super( "(,3.9.0),(3.9.0,)" );
```

Comment:
`[3.8.8,)` All three are broken and need fíxing

Reply:
Let's enumerate which version(s) to leave out:
* 3.8.7
* 3.9.0
* 4.0.0-alpha4
?

Reply:
Ah, right!

`[3,8.8,3.9.0),[3.9.1,4.0.0-alpha-4),[4.0.0-alpha-5,)`?